src/jdk.compiler/share/classes/com/sun/tools/javac/comp/InferenceContext.java

```diff
@@ -83,6 +94,14 @@ Type update(Type t) {
     Types types;
     Infer infer;
 
+    /* when an inference context (IC) is minimized, the minimized inference context (MIC) contains a
+     * proper subset of IC's inference vars (IC_IV). In other words there will be at least one inference variable T
+     * that belongs to IC_IV which doesn't belong to MIC_IV. We need the field below to, among other things,
+     * check for set membership for cases where the minimized context or any other context derived from it
+     * needs to deal with an inference variable that has been eliminated from IC_IV while minimizing it
+     */
+    InferenceContext supplementaryIC;
```

Comment:
Let's call this `next`, or `parent` ?

Reply:
I think it is more a `child` context than a `parent` one, `nested`?

Reply:
I see, yes makes sense